typing/env.ml

```diff
@@ -269,23 +281,50 @@ module TycompTbl =
                 opened
               @ rest
 
+    let add_lock lock next =
+      { current = Ident.empty; layer = Lock {lock; next} }
+
+    let rec find_all_and_locks ~mark name tbl acc =
```

Comment:
Maybe reuse this as the implementation of `find_all`? Also, I think `find_all_with_locks` would be a better name.

Reply:
We already have `IdTbl.find_all_and_locks`, so I followed the naming.
Also, "find all with locks" can be interpreted as "(find all) with locks", which is not what the function does. Adding to that confusion, this function takes `acc : locks`.

Reply:
I'm also not sure about reuse this for `find_all`, because that require iterating the returned list, which might hurt performance.